traversal/iterator/GraphIterator.java

```diff
@@ -102,8 +103,14 @@ else if (state == State.INIT) {
             }
             return state == State.FETCHED;
         } catch (Throwable e) {
-            LOG.error("Parameters: " + params.toString());
-            LOG.error("GraphProcedure: " + procedure.toString());
+            // note: catching runtime exception until we can gracefully interrupt running queries on tx close
+            if (e instanceof GraknException && ((GraknException) e).code().isPresent()
+                    && ((GraknException) e).code().get().equals(TRANSACTION_CLOSED.code())) {
+                LOG.debug("Transaction was closed during graph iteration");
+            } else {
+                LOG.error("Parameters: " + params.toString());
+                LOG.error("GraphProcedure: " + procedure.toString());
+            }
             throw e;
```

Comment:
Do you still want to throw exception if transaction was closed? Looks like from your code that's not the case? You only `LOG.error()` in the `else` block.

Reply:
it needs to propagate up to however's owning this to make sure they know the transaction is closed - might be their only signal. It just that this is so common it gets really slow and ugly to LOG.error the entire procedure each time it happens